pydantic/experimental/pipeline.py

```diff
@@ -0,0 +1,580 @@
+"""Experimental functionality, be prepared for this to break."""
+
+from __future__ import annotations
+
+import datetime
+import operator
+import re
+from collections import deque
+from collections.abc import Container
+from dataclasses import dataclass
+from decimal import Decimal
+from functools import cached_property, partial
+from typing import TYPE_CHECKING, Any, Callable, Generic, Pattern, Protocol, TypeVar, Union, overload
+
+import annotated_types
+from typing_extensions import Annotated
+
+if TYPE_CHECKING:
+    from pydantic_core import core_schema as cs
+
+    from pydantic import GetCoreSchemaHandler
+
+from pydantic._internal._internal_dataclass import slots_true
+
```

Comment:
please can we add `__all__` here and make sure we're clear about what's public and what's not.

Ideally everything that isn't public should start with an underscore.

Reply:
bed07529

Reply:
I moved it to the top